src/transactionsFactories/tokenManagementTransactionsFactory.ts

```diff
@@ -640,6 +660,231 @@ export class TokenManagementTransactionsFactory {
+        uris: string[];
+    }): Transaction {
+        if (!options.uris.length) {
+            throw new ErrBadUsage("");
+        }
+
+        const dataParts = [
+            "ESDTSetNewURIs",
+            ...this.argSerializer.valuesToStrings([
+                new StringValue(options.tokenIdentifier),
+                new BigUIntValue(options.tokenNonce),
+                ...options.uris.map((uri) => new StringValue(uri)),
+            ]),
+        ];
+
+        return new TransactionBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: options.sender,
+            dataParts: dataParts,
+            gasLimit: this.config.gasLimitSetNewUris,
+            addDataMovementGas: true,
+        }).build();
+    }
+
+    createTransactionForModifyingCreator(options: {
+        sender: IAddress;
+        tokenIdentifier: string;
+        tokenNonce: bigint;
+    }): Transaction {
+        const dataParts = [
+            "ESDTModifyCreator",
+            ...this.argSerializer.valuesToStrings([
+                new StringValue(options.tokenIdentifier),
+                new BigUIntValue(options.tokenNonce),
+            ]),
+        ];
+
+        return new TransactionBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: options.sender,
+            dataParts: dataParts,
+            gasLimit: this.config.gasLimitEsdtModifyCreator,
+            addDataMovementGas: true,
+        }).build();
+    }
+
+    createTransactionForUpdatingMetadata(options: {
+        sender: IAddress;
+        tokenIdentifier: string;
+        tokenNonce: bigint;
+        tokenName: string;
+        royalties: bigint;
+        hash: string;
+        attributes: Uint8Array;
+        uris: string[];
+    }): Transaction {
+        const dataParts = [
+            "ESDTMetaDataUpdate",
+            ...this.argSerializer.valuesToStrings([
+                new StringValue(options.tokenIdentifier),
+                new BigUIntValue(options.tokenNonce),
+                new StringValue(options.tokenName),
+                new BigUIntValue(options.royalties),
+                new StringValue(options.hash),
+                new BytesValue(Buffer.from(options.attributes)),
+                ...options.uris.map((uri) => new StringValue(uri)),
+            ]),
+        ];
+
+        return new TransactionBuilder({
+            config: this.config,
+            sender: options.sender,
+            receiver: options.sender,
+            dataParts: dataParts,
+            gasLimit: this.config.gasLimitEsdtMetadataUpdate,
+            addDataMovementGas: true,
+        }).build();
+    }
+
+    createTransactionForNftMetadataRecreate(options: {
+        sender: IAddress;
+        tokenIdentifier: string;
+        tokenNonce: bigint;
+        tokenName: string;
+        royalties: bigint;
+        hash: string;
+        attributes: Uint8Array;
+        uris: string[];
```

Comment:
Above we have a check on non-empty URIs. Should we have it here, as well?

Reply:
I'd say no.